qiskit/algorithms/phase_estimators/phase_estimator.py

```diff
@@ -32,14 +32,19 @@ class PhaseEstimator(ABC):
     @abstractmethod
     def estimate(
         self,
-        unitary: Optional[QuantumCircuit] = None,
-        state_preparation: Optional[QuantumCircuit] = None,
-        pe_circuit: Optional[QuantumCircuit] = None,
-        num_unitary_qubits: Optional[int] = None,
+        unitary: QuantumCircuit | None = None,
+        state_preparation: QuantumCircuit | None = None,
+        pe_circuit: QuantumCircuit | None = None,
+        num_unitary_qubits: int | None = None,
     ) -> "PhaseEstimatorResult":
         """Estimate the phase."""
         raise NotImplementedError
 
+    def _get_bitstring(self, length: int, number: int) -> str:
+        unpadded_bitstring = "{0:b}".format(number)[::-1]
+        padding = "0" * (length - len(unpadded_bitstring))
+        return unpadded_bitstring + padding
```

Comment:
`zfill` pads to the left. The existing code pads to the right. Too bad because zfill is more clear and faster. But I think the following will work. In a couple of tests it's 40% or so faster, and it is perhaps more clear than the original
```python
return f"{number:b}"[::-1].ljust(length, "0")
```

Reply:
But, I see this was already committed and presumably passed tests. What am I missing?

Reply:
I commited `f"{number:b}".zfill(length)[::-1]`

Reply:
Let me make it more clear: The edit that introduces `zfill` changes the output of the function. Below, `bitstring1` and `bitstring3` give the same output, while `bitstring2` gives something different.
```python
def get_bitstring1(length: int, number: int) -> str:
    unpadded_bitstring = "{0:b}".format(number)[::-1]
    padding = "0" * (length - len(unpadded_bitstring))
    return unpadded_bitstring + padding

def get_bitstring2(length: int, number: int) -> str:
    return f"{number:b}"[::-1].zfill(length)

def get_bitstring3(length: int, number: int) -> str:
    return f"{number:b}"[::-1].ljust(length, "0")
```
Then:
```python
In [2]: get_bitstring1(4, 4)
Out[2]: '0010'

In [3]: get_bitstring2(4, 4)
Out[3]: '0001'

In [4]: get_bitstring3(4, 4)
Out[4]: '0010'
```